File: rules-interpreter/src/main/java/io/github/iamrenny/ruleflow/evaluators/ParenthesisContextEvaluator.java
package io.github.iamrenny.ruleflow.evaluators;

import io.github.iamrenny.ruleflow.RuleFlowLanguageParser.ParenthesisContext;
import io.github.iamrenny.ruleflow.visitors.Visitor;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ParenthesisContextEvaluator implements ContextEvaluator<ParenthesisContext> {
    private static final Logger logger = LoggerFactory.getLogger(ParenthesisContextEvaluator.class);

    @Override
    public Object evaluate(ParenthesisContext ctx, Visitor visitor) {
        Object result = visitor.visit(ctx.expr());
        logger.debug("Parenthesis: expr={}, result={}", ctx.expr().getText(), result);
        return result;
    }
}

File: analang-interpreter/src/main/kotlin/com/rappi/fraud/rules/parser/evaluators/ParenthesisContextEvaluator.kt
package com.rappi.fraud.rules.parser.evaluators

import com.rappi.fraud.analang.ANAParser.ParenthesisContext
import com.rappi.fraud.rules.parser.visitors.Visitor

class ParenthesisContextEvaluator : ContextEvaluator<ParenthesisContext> {

    override fun evaluate(ctx: ParenthesisContext, visitor: Visitor): Any {
        return visitor.visit(ctx.expr())
    }
}